src/Models/MenuItem.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Portable\FilaCms\Models;
+
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Overtrue\LaravelVersionable\Versionable;
+use Overtrue\LaravelVersionable\VersionStrategy;
+
+class MenuItem extends Model
+{
+    use Versionable;
+    use SoftDeletes;
+    use HasFactory;
+
+    protected $versionStrategy = VersionStrategy::SNAPSHOT;
+    protected static $unguarded = true;
+
+    protected $versionable = [
+        'name',
+        'type',
+        'order',
+        'reference',
+        'parent_id',
+    ];
+
+    protected $appends = [
+        'reference_text',
+        'reference_page',
+        'reference_content'
+    ];
+
+    protected $casts = [
+        'reference' => 'json'
+    ];
+
+    public static function boot()
+    {
+        parent::boot();
+    }
+
+    public static function booted(): void
+    {
+        static::created(function (MenuItem $item) {
+            // auto-add order with end of list
+            $count = MenuItem::where('menu_id', $item->menu_id)->get()->count();
```

Comment:
I'd do `max` instead of `count` here.  They may delete items and you'll have gaps in the sequence.